src/hotspot/share/cds/metaspaceShared.cpp

```diff
@@ -751,12 +752,38 @@ void MetaspaceShared::preload_classes(TRAPS) {
     }
   }
 
-  // Exercise the manifest processing code to ensure classes used by CDS at runtime
-  // are always archived
+  // Some classes are used at CDS runtime but are not loaded, and therefore archived, at
+  // dumptime. We can perform dummmy calls to these classes at dumptime to ensure they
+  // are archived.
+  make_dummy_calls(CHECK);
+
+  log_info(cds)("Loading classes to share: done.");
+}
+
+void MetaspaceShared::make_dummy_calls(TRAPS) {
+  // Exercise the manifest processing code
```

Comment:
Nit: can we think of a better name rather than "dummy"?

Reply:
Would `exercise_runtime_cds_code()` be better?